components/maker/legal/index.js

```diff
@@ -0,0 +1,153 @@
+
+const ParticipationAgreement = ({ app, apiServer, user }) => {
+  const [name, setName] = useState('')
+  const [email, setEmail] = useState('')
+  const [embedUrl, setEmbedUrl] = useState(null)
+  const [loading, setLoading] = useState(false)
+  const [finished, setFinished] = useState(!!app.hasAcceptedSECTerms)
+  const [modalState, setModalState] = useState(false)
+  const [taxType, setTaxType] = useState(null)
+
+  const isUsa = () => taxType === 'us'
+
+  const openEverSign = url => {
+    // eslint-disable-next-line no-undef
+    eversign.open({
+      url,
+      containerID: 'embedded-participation-agreement',
+      height: '700px',
+      width: '100%',
+      events: {
+        signed: () => setFinished(true),
+        loaded: () => console.log('Eversign finished loading'),
+        declined: () => console.log('Eversign denied'),
+        error: () => console.log('Eversign error')
+      }
+    })
+  }
+
+  const getDocument = async () => {
+    setLoading(true)
+    const url = `${apiServer}/api/maker/apps/${app.id}/make-participation-agreement?name=${name}&email=${email}&isUSA=${isUsa()}&appId=${app.id}`
+    const response = await fetch(url, {
+      method: 'POST',
+      headers: new Headers({
+        authorization: `Bearer ${user.jwt}`
+      })
+    })
+    const data = await response.json()
+    setLoading(false)
+    setEmbedUrl(data.embedURL)
+    openEverSign(embedUrl)
+  }
+
+  const getDownload = async () => {
+    download(`/static/docs/${taxType}.zip`)
+    getDocument()
+    setModalState(true)
+  }
+
+  useEffect(() => {
+    if (app.eversignDocumentID) {
+      getDocument()
+    }
+  }, [app.id])
+
+  const options = [
+    { value: 'us', label: 'I am a US person or entity' },
+    { value: 'intl', label: 'I am a person based outside of the US' },
+    { value: 'intl_entity', label: 'I am an entity based outside of the US' }
+  ]
+
+  const taxStatusRadioList = options.map(({ value, label }) =>
+    <MakerRadioListLabel key={value}>
+      <input
+        type="radio"
+        name="tax-status"
+        value={value}
+        onChange={(e) => setTaxType(e.target.value)}
+      />
+      {label}
+    </MakerRadioListLabel>
+  )
+
+  return (
+    <>
+      <MakerModal isOpen={modalState} handleClose={() => setModalState(false)}>
+        <Box width={1} id="embedded-participation-agreement" />
+      </MakerModal>
+
+      <Container>
+        <MakerCardHeader>Legal Documents</MakerCardHeader>
+
+        <MakerCardText mb={4} mt={0}>
+          Please select the appropriate legal status so we can provide you with the correct Tax Form and Participation Agreement.
+        </MakerCardText>
+        <Field
+          name="name"
+          label="Full Name"
+          onChange={(e) => setName(e.target.value)}
+          value={name}
+        />
+        <Field
+          name="emailAddress"
+          type="email"
+          label="Email address"
+          onChange={(e) => setEmail(e.target.value)}
```

Comment:
should this be debounced?

Reply:
Same here, doesn't trigger anything so guess it's safe as is.